@typescript-eslint/no-unsafe-argument

**Describe the bug**

The examples fails to adhere to this typescript eslint rules
https://typescript-eslint.io/rules/no-unsafe-argument/

**To Reproduce**
Steps to reproduce the behavior:
add this `.eslintrc.json` to the root of this repository
```json
{
  "overrides": [
    {
      "extends": [
        "plugin:@typescript-eslint/recommended-requiring-type-checking"
      ],
      "files": ["*.ts", "*.tsx"],
      "parserOptions": {
        "project": "tsconfig.json"
      }
    }
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "plugins": ["@typescript-eslint"],
  "extends": ["plugin:@typescript-eslint/recommended"]
}
```
run `eslint test/integration.test.js`
There will be error:
```
  12:35  error  Unsafe argument of type `any` assigned to a parameter of type `ResponseTransformer<DefaultBodyType, any>`  @typescript-eslint/no-unsafe-argument
  15:35  error  Unsafe argument of type `any` assigned to a parameter of type `ResponseTransformer<DefaultBodyType, any>`  @typescript-eslint/no-unsafe-argument
  18:35  error  Unsafe argument of type `any` assigned to a parameter of type `ResponseTransformer<any, any>`              @typescript-eslint/no-unsafe-argument
  21:35  error  Unsafe argument of type `any` assigned to a parameter of type `ResponseTransformer<DefaultBodyType, any>`  @typescript-eslint/no-unsafe-argument
  24:35  error  Unsafe argument of type `any` assigned to a parameter of type `ResponseTransformer<DefaultBodyType, any>`  @typescript-eslint/no-unsafe-argument
  27:35  error  Unsafe argument of type `any` assigned to a parameter of type `ResponseTransformer<any, any>`              @typescript-eslint/no-unsafe-argument
```

**Expected behavior**

There should be no error.

--- a/src/createTRPCMsw.ts
+++ b/src/createTRPCMsw.ts
@@ -4,6 +4,7 @@ import {
   MockedRequest,
   PathParams,
   ResponseResolver,
+  ResponseTransformer,
   rest,
   RestContext,
   RestHandler,
@@ -93,7 +94,9 @@ const createTRPCMsw = <Router extends AnyRouter>(
   }
 
   type ContextWithDataTransformer<T extends Router[any], K extends keyof T = keyof T> = RestContext & {
-    data: (data: T[K] extends BuildProcedure<any, any, infer P> ? P : never) => any
+    data: (
+      data: T[K] extends BuildProcedure<any, any, infer P> ? P : never
+    ) => ResponseTransformer<DefaultBodyType, any>
   }
 
   type SetQueryHandler<T extends Router[any], K extends keyof T> = (

